packages/devtools_app/lib/src/code_size/code_size_screen.dart

```diff
@@ -115,6 +123,37 @@ class CodeSizeBodyState extends State<CodeSizeBody>
     }
   }
 
+  DropdownButtonHideUnderline _buildDiffTreeTypeDropdown() {
+    return DropdownButtonHideUnderline(
+      child: DropdownButton<DiffTreeType>(
+        value: _diffTreeType,
+        items: [
+          _buildMenuItem(DiffTreeType.combined),
+          _buildMenuItem(DiffTreeType.increaseOnly),
+          _buildMenuItem(DiffTreeType.decreaseOnly),
+        ],
+        onChanged: (newDiffTreeType) {
+          setState(() {
+            _diffTreeType = newDiffTreeType;
+          });
+          controller.loadFakeDiffData(
+            'old_v8.json',
+            'new_v8.json',
+            _diffTreeType,
+          );
```

Comment:
have onchanged call a method on the controller that loads the new data and sets the value of the difftreetype notifier. then wrap this DropdownButtonHideUnderline in a ValueListenableBuilder that listens to that notifier